### PR TITLE
Adds docstrings

### PR DESCRIPTION
**Pull request by Shreshtth**
- Added docstrings explaining each function.

### CODE REVIEW

File: entrypoint.py [hunk 1]
```diff
@@ -40,6 +48,14 @@ def download(download_url, path):
     help="Cutoff frames for selected/smooth type (default=9)",
 )
 def extract(path, scope, frame_cutoff):
```
Comment (onlinejudge95) [inline entrypoint.py:50]: Let's add type hints for these arguments

File: src/download.py [hunk 2]
```diff
@@ -7,6 +7,12 @@
 
 
 def setup_dataset_directory(path: str):
+    """
+    Set up the necessary directory structure for the dataset.
+
+    Args:
+        path (str): The local directory path for the dataset.
```
Comment (onlinejudge95) [inline src/download.py:14]: type hints are already provided no need to add redundant info

Comment (onlinejudge95) [reply src/download.py:14]: do it for all the functions

File: src/extract.py [hunk 3]
```diff
@@ -123,6 +161,16 @@ def _extract_smooth_images(video_file_path: pathlib.Path, frame_cutoff: int):
 def _get_frame_indices_selected(
     file_path: pathlib.Path, num_frames: int
 ) -> typing.Set[int]:
+    """
+    Get the set of frame indices to extract based on annotations.
```
Comment (onlinejudge95) [inline src/extract.py:165]: Be descriptive as to what logic is being used in terms of `num_frames`

File: src/extract.py [hunk 4]
```diff
@@ -139,6 +187,17 @@ def _get_frame_indices_selected(
 def _get_frame_indices_smooth(
     file_path: pathlib.Path, num_frames: int
 ) -> typing.Set[int]:
+    """
+    Get the set of frame indices to extract around annotated events with smoothing.
```
Comment (onlinejudge95) [inline src/extract.py:191]: Be descriptive as to what logic is being used in terms of `num_frames`

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

**Comment by Shreshtth** [comment]
@onlinejudge95 I've taken all points from the review in account. And committed the new code.

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]
@Shreshtth I have solved the merge conflict. Update the docstrings for the new functions.

**Review by onlinejudge95 — APPROVED** [review]

**Pull request merged**